aten/src/ATen/core/jit_type.h

```diff
@@ -1347,6 +1347,66 @@ matchTypeVariables(TypePtr formal, TypePtr actual, TypeEnv& type_env);
 // does not appear in `type_env`
 CAFFE2_API TypePtr tryEvalTypeVariables(TypePtr type, TypeEnv& type_env);
 
+struct InterfaceType;
```

Comment:
the reason of it being moved is because classType::is_module() need to know interfaceType so that it knows the submember is a module or not. I can do forward declaration for InterfaceType but this might be more neat.